src/core_plugins/tile_map/public/editors/wms_options.js

```diff
@@ -27,32 +27,28 @@ module.directive('wmsOptions', function (serviceSettings) {
     template: wmsOptionsTemplate,
     replace: true,
     scope: {
-      options: '='
+      options: '=',
+      collections: '=',
     },
     link: function ($scope) {
 
-      $scope.options.baseLayersAreLoaded = new Promise((resolve, reject) => {
+      new Promise((resolve, reject) => {
 
         serviceSettings
           .getTMSServices()
           .then((allTMSServices) => {
-
-            if (!$scope.options.tmsLayers) {
-              $scope.options.tmsLayers = [];
-            }
-
-            const newBaseLayers = $scope.options.tmsLayers.slice();
+            const newBaseLayers = $scope.collections.tmsLayers.slice();
             for (let i = 0; i < allTMSServices.length; i += 1) {
               const layerFromService = allTMSServices[i];
               const alreadyAdded = newBaseLayers.some((layer) => layerFromService.id === layer.id);
               if (!alreadyAdded) {
                 newBaseLayers.push(layerFromService);
               }
             }
-            $scope.options.tmsLayers = newBaseLayers;
+            $scope.collections.tmsLayers = newBaseLayers;
 
-            if (!$scope.options.selectedTmsLayer) {
-              $scope.options.selectedTmsLayer = $scope.options.tmsLayers[0];
+            if (!$scope.options.selectedTmsLayer && $scope.collections.tmsLayers.length) {
+              $scope.options.selectedTmsLayer = $scope.collections.tmsLayers[0];
```

Comment:
When this happens editor state will go to dirty. I think this is ok, as this will only happen when you create a new visualization. You will setup the aggregations anyway and press the play button and at that time this would get applied as well.

If that is not sufficient we could also call `stageEditableVis` here, which will give control back to visualization (another `render()` call)

Reply:
if this base layer no longer exists then then the base layer won't show up (on dashboard)
going into edit mode the base layer also won't show up and the base layer dropdown will be unselected